feature/profilecard/src/commonMain/kotlin/io/github/droidkaigi/confsched/profilecard/ProfileCardScreenPresenter.kt

```diff
@@ -17,54 +17,63 @@ internal sealed interface ProfileCardScreenEvent
 
 internal sealed interface EditScreenEvent : ProfileCardScreenEvent {
     data object SelectImage : EditScreenEvent
-    data class Create(val profileCard: ProfileCard) : EditScreenEvent
+    data class Create(val profileCard: ProfileCard.Exists) : EditScreenEvent
 }
 
 internal sealed interface CardScreenEvent : ProfileCardScreenEvent {
     data object Share : CardScreenEvent
     data object Edit : CardScreenEvent
 }
 
-private fun ProfileCard?.toEditUiState(): ProfileCardUiState.Edit {
-    if (this == null) {
-        return ProfileCardUiState.Edit()
+private fun ProfileCard.toEditUiState(): ProfileCardUiState.Edit {
+    return when (this) {
+        is ProfileCard.Exists -> ProfileCardUiState.Edit(
+            nickname = nickname,
+            occupation = occupation,
+            link = link,
+            image = image,
+            theme = theme,
+        )
+        ProfileCard.DoesNotExists, ProfileCard.Loading -> ProfileCardUiState.Edit()
     }
-    return ProfileCardUiState.Edit(
-        nickname = nickname,
-        occupation = occupation,
-        link = link,
-        image = image,
-        theme = theme,
-    )
 }
 
-private fun ProfileCard.toCardUiState(): ProfileCardUiState.Card {
-    return ProfileCardUiState.Card(
-        nickname = nickname,
-        occupation = occupation,
-        link = link,
-        image = image,
-        theme = theme,
-    )
+private fun ProfileCard.toCardUiState(): ProfileCardUiState.Card? {
+    return when (this) {
+        is ProfileCard.Exists -> ProfileCardUiState.Card(
+            nickname = nickname,
+            occupation = occupation,
+            link = link,
+            image = image,
+            theme = theme,
+        )
+        ProfileCard.DoesNotExists, ProfileCard.Loading -> null
+    }
 }
 
 @Composable
 internal fun profileCardScreenPresenter(
     events: Flow<ProfileCardScreenEvent>,
     repository: ProfileCardRepository = localProfileCardRepository(),
 ): ProfileCardScreenState = providePresenterDefaults { userMessageStateHolder ->
-    val profileCard: ProfileCard? by rememberUpdatedState(repository.profileCard())
+    val profileCard: ProfileCard by rememberUpdatedState(repository.profileCard())
     var isLoading: Boolean by remember { mutableStateOf(false) }
-    val ediUiState: ProfileCardUiState.Edit by rememberUpdatedState(profileCard.toEditUiState())
-    val cardUiState: ProfileCardUiState.Card? by rememberUpdatedState(profileCard?.toCardUiState())
-    var uiType: ProfileCardUiType by remember { mutableStateOf(ProfileCardUiType.Edit) }
+    val editUiState: ProfileCardUiState.Edit by rememberUpdatedState(profileCard.toEditUiState())
+    val cardUiState: ProfileCardUiState.Card? by rememberUpdatedState(
+        if (profileCard is ProfileCard.Exists) {
+            profileCard.toCardUiState()
+        } else {
+            null
+        },
+    )
```

Comment:
I think if-expression is not needed.

```suggestion
    val cardUiState: ProfileCardUiState.Card? by rememberUpdatedState(profileCard.toCardUiState())
```